src/main/helpers/controller/ClaimantEmploymentDatesEnterHelper.ts

```diff
@@ -0,0 +1,43 @@
+import { Response } from 'express';
+
+import { AppRequest } from '../../definitions/appRequest';
+import { CaseDate } from '../../definitions/case';
+import { PageUrls, ValidationErrors } from '../../definitions/constants';
+import ErrorUtils from '../../utils/ErrorUtils';
+import { isDateEmpty, isDateInputInvalid, isFirstDateBeforeSecond } from '../../validators/dateValidators';
+import { setUrlLanguage } from '../LanguageHelper';
+
+export const isEndDateBeforeStartDate = (req: AppRequest): boolean => {
+  const startDate: CaseDate = {
+    day: req.body['et3ResponseEmploymentStartDate-day'],
+    month: req.body['et3ResponseEmploymentStartDate-month'],
+    year: req.body['et3ResponseEmploymentStartDate-year'],
+  };
+
+  const endDate: CaseDate = {
+    day: req.body['et3ResponseEmploymentEndDate-day'],
+    month: req.body['et3ResponseEmploymentEndDate-month'],
+    year: req.body['et3ResponseEmploymentEndDate-year'],
+  };
+
+  if (isDateEmpty(startDate) && isDateEmpty(endDate)) {
+    return false;
+  }
+
+  if (isDateInputInvalid(startDate) || isDateInputInvalid(endDate)) {
```

Comment:
isn't this calling the same method isDateEmpty() as in the if condition on line 23? If you are checking start OR end dates for non-empty, do you really need to check start AND end dates for non-empty if that makes sense?